### ISSUE TITLE
Can't start selfhost build

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#123072]
```
The terminal process "/bin/zsh '-c', 'yarn run watch-extensionsd'" failed to launch (exit code: 127).
```

I get this when running the build task from VSCode insiders (`Cmd+Shift+B`).

![image](https://user-images.githubusercontent.com/900690/117249152-69703280-ae41-11eb-8ff8-c02ea6ca912c.png)

**Comment by bpasero** [comment]
Looks like a regression from the removal of shell env in renderer and/or shared process, does not reproduce when running from a terminal.

**Comment by joaomoreno** [comment]
The same happens to me on Windows + WSL.

**Comment by Tyriar** [comment]
Yeah I'm hitting this too and just started looking into it. It is indeed because of the shell environment patching. @alexr00 what do you think about having tasks run login shells by default on macOS?

**Comment by joaomoreno** [comment]
@Tyriar Note that the same happens to me on Windows + WSL.

**Comment by Tyriar** [comment]
@joaomoreno thanks for the info. An alternative maybe safer approach is for tasks to pass in the patched environment in `IShellLaunchConfig.env`

**Comment by Tyriar** [comment]
Here's my plan for this:

- Bring back `terminal.integrated.inheritEnv`, keep the default as `true` which will use the patched environment and `false` we will use the unpatched environment (instead of the solution which had a bunch of issues before).
- Add `IShellLaunchConfig.inheritEnv?: boolean` which will let tasks override whatever the setting is since they could break otherwise.
- For WSL (maybe other remotes) terminals we don't run a login shell on start up, so things on `$PATH` may be missing in regular terminals when `terminal.integrated.inheritEnv` is set to `false`. This will be considered by design and it will be up to the user to run their terminal as a login shell if that's desired, I'll explain this in the `terminal.integrated.inheritEnv` config description.

**Comment by reu-res** [comment]
Argument handling problem, commas are added after each argument.
```json
// See https://go.microsoft.com/fwlink/?LinkId=733558
// for the documentation about the tasks.json format
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "npm",
      "script": "watch",
      "problemMatcher": "$tsc-watch",
      "isBackground": true,
      "presentation": {
        "reveal": "never"
      },
      "group": {
        "kind": "build",
        "isDefault": true
      },
      "label": "npm: watch",
      "detail": "tsc -watch -p ./",
      "options": {
        "shell": {
          "executable": "/bin/bash",
          "args": ["-c", "-i"]
        }
      }
    }
  ]
}
```
<img width="472" alt="Screenshot 2021-05-06 at 18 27 28" src="https://user-images.githubusercontent.com/1050186/117325380-55581f80-ae99-11eb-9393-9a17421b43c8.png">

**Comment by Tyriar** [comment]
@reu-res that's just the notification's format, the underlying issue is for a different reason.

**Comment by joaomoreno** [comment]
I still get this issue in today's Insiders. Toggling `terminal.integrated.inheritEnv` makes no difference at all. 😢

**Comment by reu-res** [comment]
For me now everything works without problems on macOS. Regardless of the value of the variable `terminal.integrated.inheritEnv`

**Comment by Tyriar** [comment]
@reu-res it also depended on whether you launch VS Code from the terminal or the dock, maybe that's it?

**Comment by Tyriar** [comment]
@joaomoreno works for me, any more info? Trace logging the terminal launch would be useful so we can see how it gets spawned, you should see this in the shared proc window:

<img width="419" alt="Screen Shot 2021-05-07 at 4 55 22 AM" src="https://user-images.githubusercontent.com/2193314/117446092-6f711c80-aef0-11eb-93a8-186d1091301d.png">

If you have inheritEnv set to true it should have `$SHLVL` set

**Comment by reu-res** [comment]
> @reu-res it also depended on whether you launch VS Code from the terminal or the dock, maybe that's it?

It doesn't matter. Works in both cases. Today's build works regardless of the value of the inheritEnv variable.

**Comment by Tyriar** [comment]
I'm blocked on this without more info on a repro.

**Comment by joaomoreno** [comment]
![image](https://user-images.githubusercontent.com/22350/117501092-e1dc0e00-af7d-11eb-9b72-e0006e045964.png)

I'm using zsh and oh-my-zsh. For node and yarn, I'm using asdf. The only configuration I've done to add asdf to zsh was to add its plugin like so:

```
➜  ~ cat .zshrc | grep asdf
plugins=(git asdf)
```

Maybe the way we spawn zsh in Code skips loading oh-my-zsh plugins?

**Comment by jasonwilliams** [comment]
> I'm blocked on this without more info on a repro.

https://github.com/microsoft/vscode/issues/123209

**Comment by Tyriar** [comment]
@jasonwilliams do you have any terminal settings set?

**Comment by jasonwilliams** [comment]
> @jasonwilliams do you have any terminal settings set?

@Tyriar I don't have any terminal settings set

**Comment by jasonwilliams** [comment]
The below is on:
Version: 1.57.0-insider (system setup)
Commit: a19565da7dfc739fa26e36aae96113e19fe3806f
Date: 2021-05-07T17:19:03.737Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.19042

![cargoNotFound2](https://user-images.githubusercontent.com/936006/117536462-0de5a680-aff3-11eb-9bff-3106bc3a5215.gif)

**Comment by joaomoreno** [comment]
@jasonwilliams I believe that build didn't contain the fix yet. We just released another Insiders which contains the fix, my issue is gone:

```
Version: 1.57.0-insider (user setup)
Commit: 37a12c9b0846167b2f68e9e1f276149d2d8b9b27
Date: 2021-05-12T14:14:26.024Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.19043
```

**Comment by jasonwilliams** [comment]
I can confirm it’s working also, thanks!
